subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -2070,6 +2072,11 @@ public FileCollection getArtifactFiles() {
             return result.get().artifactResults;
         }
 
+        @Override
+        public Provider<Set<ResolvedArtifactResult>> getResolvedArtifacts() {
+            return new BuildableBackedSetProvider<>(getArtifactFiles(), new ArtifactCollectionResolvedArtifactsFactory(this));
```

Comment:
🟡 reuse `SerializableLambdas.factory`?

`         return new BuildableBackedSetProvider<>(getArtifactFiles(), factory(this::getArtifacts));`

Reply:
This makes a bunch of tests fail with:
```
Caused by: java.lang.ClassCastException: class org.gradle.configurationcache.serialization.codecs.FixedArtifactCollection cannot be cast to class org.gradle.api.internal.artifacts.configurations.DefaultConfiguration$ConfigurationArtifactCollection (org.gradle.configurationcache.serialization.codecs.FixedArtifactCollection and org.gradle.api.internal.artifacts.configurations.DefaultConfiguration$ConfigurationArtifactCollection are in unnamed module of loader org.gradle.initialization.MixInLegacyTypesClassLoader @7fd7a283)
	at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration$ConfigurationArtifactCollection.$deserializeLambda$(DefaultConfiguration.java:2043)
	... 184 more
```
It may be due to the fact that we are in an inner class. I'll keep the explicit class for now.

Reply:
Hm, odd.